### PR TITLE
fix: improve Cloudflare Pages preview deletion pagination logic

### PR DESCRIPTION
**Pull request by amritk**
## Problem

The script for deleting Cloudflare Pages preview deployments had several issues:

1. **Unreliable pagination**: It relied on `result_info.total_pages` from the API, which Cloudflare doesn't always populate, making pagination logic fragile.
2. **Poor error handling**: The script would exit immediately on the first deletion failure, preventing cleanup of other deployments.
3. **Limited test coverage**: Tests didn't cover edge cases like partial pages, no deployments found, or individual deletion failures.

## Solution

**Pagination improvements:**
- Changed from relying on `total_pages` to detecting partial pages (when result count < per_page size)
- This is more robust since Cloudflare doesn't always include `total_pages` in responses
- Added configurable `CLOUDFLARE_PER_PAGE` environment variable (defaults to 100)

**Error handling improvements:**
- Changed deletion failures from immediately exiting to continuing with remaining deployments
- Tracks failed deletion count and reports summary at the end
- Script still exits with code 1 if any deletions failed, but attempts all deployments first

**Test coverage:**
- Updated existing test to verify pagination works without relying on `total_pages`
- Added test for stopping pagination when first page is partial
- Added test for handling branches with no deployments
- Added test for continuing after individual deletion failures
- Renamed "reports API errors" test to clarify it tests listing failures

**Output improvements:**
- Changed page output from "page X of Y" to just "page X" (since Y is no longer reliable)

## Checklist

- [x] I explained why the change is needed.
- [ ] I added a changeset. 
- [x] I added tests.
- [ ] I updated the documentation. 

https://claude.ai/code/session_01NPYZKqqDB2n9PkCp9jFtAb

---

> [!NOTE]
> **Medium Risk**
> Moderate risk because it changes pagination termination and failure handling for destructive Cloudflare API deletes; mistakes could leave previews undeleted or attempt more API calls than expected.
> 
> **Overview**
> Improves the `delete-cloudflare-pages-preview.sh` cleanup script to paginate without relying on `result_info.total_pages`, instead stopping when a page returns fewer than `CLOUDFLARE_PER_PAGE` results (now configurable). It also hardens API error reporting, reuses a single temp workspace, and changes deletion behavior to continue through all deployments while tracking failures and exiting non-zero only after attempting everything.
> 
> Updates and expands tests to cover missing `total_pages`, partial-page pagination termination, no matching deployments, and continuing after individual deletion failures.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b6f1e97b337deff2817134c7ac0abc317fb9472f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tooling/scripts/delete-cloudflare-pages-preview.sh [hunk 1]
```diff
@@ -37,8 +37,15 @@ while [ "$page" -le "$total_pages" ]; do
...
     "$response_file" >> "$deployment_ids_file"
 
-  total_pages="$(jq -r '.result_info.total_pages // 1' "$response_file")"
-  echo "Scanned Cloudflare deployment page $page of $total_pages"
+  result_count="$(jq -r '.result // [] | length' "$response_file")"
+  echo "Scanned Cloudflare deployment page $page"
+
+  # Stop once we receive a partial page. Cloudflare's listing endpoint does not
+  # always populate result_info.total_pages, so we paginate by page size instead.
+  if [ "$result_count" -lt "$per_page" ]; then
```
Comment (DemonHa) [inline tooling/scripts/delete-cloudflare-pages-preview.sh:95]: Do they respond with an empty list if page > total pages?

**Review by DemonHa — APPROVED** [review]
Looks good, when do we run this script tho? Should we also run on a new commit so there's only only preview per PR?

File: tooling/scripts/delete-cloudflare-pages-preview.sh (hunk 1 above)
Comment (amritk) [reply tooling/scripts/delete-cloudflare-pages-preview.sh:95]: result_info is optional LOL why cloudflare why
